Source/Applications/openHistorian/openHistorian/Startup.cs

```diff
@@ -152,6 +152,15 @@ void Register(StartupEvent startupEvent) =>
             // Enable failover requests
             app.UseFailover(ServiceHost.FailOverRequestPath);
 
+            // Add Missing Content-Securtiy Policies
+            app.Use(async (context, next) =>
+            {
+                context.Response.Headers.Add("Content-Security-Policy", ["default-src: 'self'"]);
+                context.Response.Headers.Add("Strict-Transport-Security", ["max-age=31536000","includeSubDomains"]);
+                context.Response.Headers.Add("X-Content-Type-Options", ["nosniff"]);
+                await next();
```

Comment:
I was thinking it might be better to set the response headers on the way out, so we know they will always be applied. It might also be best to invoke `app.Use()` as early in the pipeline as possible so we know that these headers will be added at the very end of processing.
```suggestion
                await next();
                context.Response.Headers.Add("Content-Security-Policy", ["default-src: 'self'"]);
                context.Response.Headers.Add("Strict-Transport-Security", ["max-age=31536000","includeSubDomains"]);
                context.Response.Headers.Add("X-Content-Type-Options", ["nosniff"]);
```

Similar to how this sort of thing is done in openXDA.
https://github.com/GridProtectionAlliance/openXDA/blob/560e8d0bb848705082159e42a3bf3072f1f59b08/Source/Applications/openXDA/openXDA/WebHosting/XDAWebHost.cs#L348-L353